models/src/main/scala/coop/rchain/models/rholangn/parmanager/ConnectiveUsed.scala

```diff
@@ -0,0 +1,61 @@
+package coop.rchain.models.rholangn.parmanager
+
+import coop.rchain.models.rholangn._
+
+private[parmanager] object ConnectiveUsed {
+  private def cUsed(p: RhoTypeN): Boolean                               = p.connectiveUsed
+  private def cUsed(kv: (RhoTypeN, RhoTypeN)): Boolean                  = cUsed(kv._1) || cUsed(kv._2)
+  private def cUsed(ps: Seq[RhoTypeN]): Boolean                         = ps.exists(cUsed)
+  private def cUsedKVPairs(kVPairs: Seq[(RhoTypeN, RhoTypeN)]): Boolean = kVPairs.exists(cUsed)
+  private def cUsed(pOpt: Option[RhoTypeN]): Boolean =
+    if (pOpt.isDefined) cUsed(pOpt.get) else false
```

Comment:
Functions with Option arguments should not be used.

Reply:
I’ve change it for most cases, but I save implementation for private functions for hashing and serialization.  Because it makes the code much clearer. We can change it later.